infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -142,9 +326,149 @@ private boolean registryTableHasNewEntries() {
+                                    subscriber, groupName);
+                        }
+                    });
+
+                    return txn.commit();
+                } catch (TransactionAbortedException tae) {
+                    log.error("Failed to preprocess client configuration tables due to Txn Abort", tae);
+                    throw new RetryNeededException();
+                }
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("Unable to preprocess client configuration tables", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Callback method which is invoked when the client config listener receives update for new client registration.
+     */
+    public void onNewClientRegister(ReplicationSubscriber subscriber) {
+        if (registeredSubscribers.contains(subscriber)) {
+            log.warn("Client {} with model {} already registered!", subscriber.getClientName(), subscriber.getModel());
+            return;
+        }
+        // TODO (V2): Currently we add a default subscriber for logical group use case instead of listening on client
+        //  registration. Subscriber should be added upon registration after grpc stream for session creation is added.
+        registeredSubscribers.add(subscriber);
+    }
+
+
+    /**
+     * Callback method which is invoked when the client config listener receives update for adding/removing
+     * Sinks from a logical group.
+     *
+     * @param subscriber   Registered replication subscriber that changes the group destination mapping.
+     * @param logicalGroup Logical group that is being modified.
+     * @param destinations The updated list of destinations that this logicalGroup is targeting to.
+     * @return A set of LogReplicationSession that needs a forced snapshot sync
+     */
+    public Set<LogReplicationSession> onGroupDestinationsChange(ReplicationSubscriber subscriber,
+                                                                String logicalGroup,
+                                                                List<String> destinations) {
+        if (!registeredSubscribers.contains(subscriber)) {
+            log.error("Client {} has not registered with replication model {}. Please register first.",
+                    subscriber.getClientName(), subscriber.getModel());
+            return null;
+        }
+
+        Set<String> recordedSinks = groupSinksMap.getOrDefault(logicalGroup, new HashSet<>());
+        Set<String> updatedSinks = new HashSet<>(destinations);
+
+        // The impacted sinks are those who are newly added to this logicalGroup or removed from it. Their config
+        // needs to be generated again to reflect the group destination change.
+        Set<String> impactedSinks = Stream.concat(recordedSinks.stream(), updatedSinks.stream())
```

Comment:
this can be simplified to get the newly added and removed sink clusters, which are needed later in the method.  Please simplify this so that !recordedSinks.contains(sink) does not need to be computed again.